packages/core/crates/core-types/src/channels.rs

```diff
@@ -734,7 +812,7 @@ pub fn win_prob_to_f64(encoded_win_prob: &EncodedWinProb) -> f64 {
 
 /// Encodes [0.0f64, 1.0f64] to [0x00000000000000, 0xffffffffffffff]
 pub fn f64_to_win_prob(win_prob: f64) -> Result<EncodedWinProb> {
-    if win_prob > 1.0 || win_prob < 0.0 {
+    if !(0.0..=1.0).contains(&win_prob) {
```

Comment:
Nice syntax, I learnt something new!. 🍻

Reply:
Most importantly, this does not short circuit like `||` - which is a bug. It should've been `&&`